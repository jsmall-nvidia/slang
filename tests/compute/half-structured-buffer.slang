//TEST(compute):COMPARE_COMPUTE:-dx12 -compute -use-dxil -profile cs_6_2
//DISABLE_TEST(compute):COMPARE_COMPUTE:-vk -compute -profile cs_6_2
//TEST_INPUT:ubuffer(data=[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], stride=16):dxbinding(0),glbinding(0),out

struct Thing
{
    uint pos;
    half2 color;
    int b;
    float radius;
};

RWStructuredBuffer<Thing> outputBuffer;

[numthreads(4, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    uint tid = dispatchThreadID.x;
    
    float v = tid * (1.0f / 3.0f);
    
    Thing thing;
    thing.pos = tid;
    thing.color = half2(v, v);
    thing.b = -(int)tid;
    
    //thing.color = half4(v, 0, 0, v);
    thing.radius = v;
    
    outputBuffer[tid] = thing;
}