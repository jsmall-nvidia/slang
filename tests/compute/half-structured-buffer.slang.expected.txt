0
0
0
0
1
3555
FFFFFFFF
3EAAAAAB
2
0
FFFFFFFE
3F2AAAAB
3
0
FFFFFFFD
3F800000
